### DISCUSSION TITLE
Locale get from API

### DISCUSSION THREAD

**Discussion by siamak** [vinissimus/next-translate#968] [General]
Is there any way to use locales json data from Server? and use fallback json from code.

For example, My server api send this as locales to my app
`{
    "translations": {
        "en": {
            "note": "Note",
            "provider": "Provider",
            "tags": "Tags"
        }
    }
}`

Also, I would like to use fallback from my code.
`{
    "translations": {
        "en": {
            "note": "Note",
            "provider": "Provider",
            "tags": "Tags",
            "reviews": "Reviews",
}
    }
}`

**Comment by aralroca** [comment]
Yes, you need to overwrite the `loadLocaleFrom` in the configuration.

i18n.js

```js
{
  // ...rest of config
  "loadLocaleFrom":  async (lang, ns) => {
      const res = await fetch('...')
      return res.json()
   }
}
```

**Reply by siamak** [reply]
Is there any way to pass them on `getStaticProps` ? @aralroca

**Reply by aralroca** [reply]
This `loadLocaleFrom` is going to be injected automatically by the next-translate webpack plugin inside the `getStaticProps`. You don't need to duplicate this logic on every page 😊

**Reply by siamak** [reply]
> This `loadLocaleFrom` is going to be injected automatically by the next-translate webpack plugin inside the `getStaticProps`. You don't need to duplicate this logic on every page 

In my case, I need to pass translations from getStaticProps. Do you have any solution for this? 

@aralroca

**Reply by aralroca** [reply]
Only to understand, why do you need to load the translations inside the page `getStaticProps` loader? What doesn't let you move this logic into `loadLocaleFrom`?

**Reply by siamak** [reply]
> Only to understand, why do you need to load the translations inside the page `getStaticProps` loader? What doesn't let you move this logic into `loadLocaleFrom`?

I have many logics for authentication on API requests. I can't move them on `i18n.js`

**Reply by aralroca** [reply]
In this case, as a workaround you can return the property `__namespaces` in the `getStaticProps` with the translations, that is what the `loadLocaleFrom` do under the hood. Let us know if it works.

It may not work because it is messing with the `__namespaces` property set by the Webpack plugin. If so, open an issue or PR and we can make a fix to respect the `__namespaces` if it is already defined or solve this with a better approach.

**Reply by siamak** [reply]
<img width="861" alt="image" src="https://user-images.githubusercontent.com/3109082/211872317-a6fbfc8c-0709-4057-bfcd-d67ee750dfd1.png">

Also, I've changed `loader: false`

It's not working.

**Reply by mattvb91** [reply]
did you manage to fix this? I need to set the locale inside getInitialProps because im loading data from an api

**Reply by mattvb91** [reply]
@aralroca im having the same issue, im passing the `__namespaces` in from getInitialProps but translations dont work. Is there a basic example of this working with getInitialProps?
